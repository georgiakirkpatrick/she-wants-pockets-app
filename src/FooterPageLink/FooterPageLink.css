.Footer__page-link {
    color: white;
}