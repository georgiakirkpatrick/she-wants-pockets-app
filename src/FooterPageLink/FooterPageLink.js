import React from 'react';
import './FooterPageLink.css'

function FooterPageLink(props) {
    return (
        <li className='Footer__page-link' key={props.id}>
            <a href={props.route}>
                {props.name}
            </a>
        </li>
    )
}

export default FooterPageLink;