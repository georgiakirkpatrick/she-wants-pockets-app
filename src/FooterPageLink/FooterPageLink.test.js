import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FooterPageLink from './FooterPageLink';

const testLink = [
    {
        id: 1,
        name: 'About',
        route: '/about'
    }
]

describe('FooterPageLink', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')

        ReactDOM.render(<FooterPageLink 
            key={testLink.id}
            route={testLink.route}
            name={testLink.name}
        />, div)

        ReactDOM.unmountComponentAtNode(div)
    });

    it ('renders the UI as expected', () => {
        const tree = renderer
            .create(<FooterPageLink 
                key={testLink.id}
                route={testLink.route}
                name={testLink.name}
            />)
            .toJSON()
            
        expect(tree).toMatchSnapshot()
    });
});    