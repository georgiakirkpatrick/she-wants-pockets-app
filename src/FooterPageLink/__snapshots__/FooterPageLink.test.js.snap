// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterPageLink renders the UI as expected 1`] = `
<li
  className="Footer__page-link"
>
  <a />
</li>
`;
